File: Data Structures & Algorithms/longest-common-prefix/submission-3.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        for i in range(len(strs[0])):
            comp = strs[0][i]
            for s in strs:
                if i >= len(s) or s[i] != comp:
                    return s[:i]
            
        return strs[0]

File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        map<array<int, 26>, vector<string>> freqmap;
        for(string word : strs) {
            array<int, 26> freq = {};
            for(char ch : word) {
                freq[int(ch) - int('a')]++;
            }
            freqmap[freq].push_back(word);
        }

        vector<vector<string>> ans;
        for(pair<array<int, 26>, vector<string>> p : freqmap) 
            ans.push_back(p.second);
        return ans;
    }
};